libutils/sequence.c

```diff
@@ -449,3 +451,114 @@ int SeqStringLength(Seq *seq)
 
     return total_length;
 }
+
+// TODO: These static helper functions could be (re)moved
+static bool HasNulByte(const char *str, size_t n)
+{
+    for (int i = 0; i < n; ++i)
+    {
+        if (str[i] == '\0')
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+static long GetLengthPrefix(const char *data)
+{
+    if (HasNulByte(data, 10))
+    {
+        return -1;
+    }
+
+    if (!isdigit(data[0]))
+    {
+        return -1;
+    }
+
+    if (data[SEQ_PREFIX_LEN-1] != ' ')
+    {
+        return -1;
+    }
+
+    // NOTE: This uses unsigned long because HPUX sscanf doesn't support %zu
```

Comment:
Why not just use `long` here when it's type-casted to `long` at the end anyway?

Reply:
Thanks! I thought `%lu` wouldn't match negative numbers, but it does :(